module/motion/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,286 @@
+
+                // compute the pitch offset to the currently wanted pitch of the engine
+                float wanted_pitch =
+                    params.trunk_pitch + params.trunk_pitch_p_coef_forward * walk_engine.getFootstep().getNext().x()
+                    + params.trunk_pitch_p_coef_turn * std::abs(walk_engine.getFootstep().getNext().z());
+                RPY.y() += wanted_pitch;
+
+                // threshold pitch and roll
+                if (std::abs(RPY[0]) > config.imu_roll_threshold) {
+                    log<NUClear::WARN>(fmt::format("Robot roll exceeds threshold - {} > {}",
+                                                   std::abs(RPY.x()),
+                                                   config.imu_roll_threshold));
+                    walk_engine.requestPause();
+                }
+                else if (std::abs(RPY[1]) > config.imu_pitch_threshold) {
+                    log<NUClear::WARN>(fmt::format("Robot pitch exceeds threshold - {} > {}",
+                                                   std::abs(RPY.y()),
+                                                   config.imu_pitch_threshold));
+                    walk_engine.requestPause();
+                }
+            }
+        });
+
+        on<Trigger<EnableWalkEngineCommand>>().then([this](const EnableWalkEngineCommand& command) {
+            subsumptionId = command.subsumptionId;
+            walk_engine.reset();
+            update_handle.enable();
+        });
+
+        on<Trigger<DisableWalkEngineCommand>>().then([this] { update_handle.disable(); });
+
+        update_handle = on<Every<UPDATE_FREQUENCY, Per<std::chrono::seconds>>, Single>().then([this]() {
+            const float dt = getTimeDelta();
+
+            if (falling) {
+                // We are falling, reset walk engine
+                walk_engine.reset();
+            }
+            else {
+                // we don't want to walk, even if we have orders, if we are not in the right state
+                bool walkableState = true;  // TODO: what is our equivalent here?
+
+                // see if the walk engine has new goals for us
+                walkableState = walk_engine.updateState(dt, current_orders, walkableState);
+
+                if ((walk_engine.getState() != engine::WalkEngineState::IDLE) && walkableState) {  // todo
+                    calculateJointGoals();
+                }
+            }
+        });
+    }  // namespace motion
+
+    float QuinticWalk::getTimeDelta() {
+        // compute time delta depended if we are currently in simulation or reality
+        float dt;
+        auto current_time = NUClear::clock::now();
+        // only take real time difference if walking was not stopped before
+        // TODO Is this really necessary?
+        auto time_diff_ms = std::chrono::duration_cast<std::chrono::milliseconds>(current_time - last_update_time);
+        dt                = time_diff_ms.count() / 1000.0f;
```

Comment:
```suggestion
        auto current_time = NUClear::clock::now();
        // only take real time difference if walking was not stopped before
        // TODO Is this really necessary?
        float dt = std::chrono::duration_cast<std::chrono::milliseconds>(current_time - last_update_time).count() / 1000.0f;
```

Reply:
Would making this change nullify the TODO above it?

Reply:
No, I think the TODO is about the calculation itself